driver/xrt/include/accl.hpp

```diff
@@ -124,6 +124,14 @@ class ACCL {
    */
   void deinit();
 
+
+  /**
+   * Initializes ACCL 
+  */
+  void initialize(const std::vector<rank_t> &ranks, int local_rank,
```

Comment:
by moving initialization to a separate function from the constructor, we run the risk of users calling into an un-initialized accl object. I can't figure out if you're handling this in any way.

Reply:
The config_rdy prevents users from performing ACCL operations without initialization (only CONFIG operations can be executed). I will modify the error message to make this even clearer